### PR TITLE
Gui Performance - Vue3 Composition API

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Implements vue-composition API.
Refactor Confguration View's Forms

# Impacts
npm packages
vue runtime

# Delete branch after merge
YES 

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* GUI configuration form refactor

## Enhancements
* ...

## Bug Fixes
* ...

### CODE REVIEW

**Comment by nqb** [comment]
I just ran a pipeline to see if build passed on CI: https://gitlab.com/inverse-inc/packetfence/-/pipelines/229502861

File: html/pfappserver/root/static.alt/README.md [hunk 1]
```diff
@@ -57,6 +56,7 @@ Since this is a SPA (Single-Page Application), it is currently accessible from `
  * [ace](https://ace.c9.io/)
 * [d3-force](https://github.com/d3/d3-force) -- d3 force directed graphs
 * [typeface-b612-mono](https://www.npmjs.com/package/typeface-b612-mono) -- small fonts
+* [mime-type](https://github.com/jshttp/mime-types) -- mime/content-type
```
Comment (cgx) [inline html/pfappserver/root/static.alt/README.md:59]: Add a reference to `yup`.

File: html/pfappserver/root/static.alt/package.json [hunk 2]
```diff
@@ -55,8 +58,9 @@
     "stylelint-config-idiomatic-css": "^1.1.0",
     "svg-arc-to-cubic-bezier": "^3.2.0",
     "vue-cli-plugin-bootstrap-vue": "0.6.0",
-    "vue-cli-plugin-i18n": "^1.0.1",
-    "vue-template-compiler": "2.6.12"
+    "vue-cli-plugin-i18n": "^0.6.1",
```
Comment (cgx) [inline html/pfappserver/root/static.alt/package.json]: Is downgrading `vue-cli-plugin-i18n` necessary?

File: html/pfappserver/root/static.alt/package.json [hunk 3]
```diff
@@ -55,8 +58,9 @@
...
+    "vue-template-compiler": "2.6.12",
+    "webpack-bundle-analyzer": "^4.2.0"
```
Comment (cgx) [inline html/pfappserver/root/static.alt/package.json:63]: `webpack-bundle-analyzer` is already a dependency of `@vue/cli-service`. You can remote it.

File: html/pfappserver/root/static.alt/src/components/new/BaseContainerLoading.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+// @vue/component
+export default {
+  name: 'base-container-loading',
+  inheritAttrs: false,
+  props,
+  setup
+}
+</script>
+,
+    default: true
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/new/BaseContainerLoading.vue]: Are those two lines necessary?

File: html/pfappserver/root/static.alt/src/composables/useFormButtonBar.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    return 'close' in listeners
+  })
+
+  const canDelete = computed(() => {
+    return isDeletable.value && isClone.value === false && isNew.value === false && 'remove' in listeners
+  })
+
+  const canReset = true /*computed(() => {
+    return isNew.value === false
+  })*/
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/composables/useFormButtonBar.js]: Unsure about the reset button? If it makes sense to always have it, remove this code.

File: html/pfappserver/root/static.alt/src/components/pfSearchBoolean.vue [hunk 6]
```diff
@@ -3,27 +3,27 @@
...
-      <b-form-select class="mr-1" v-model.lazy="model.values[0].values[0].value" :options="values(model.values[0].values[0])" v-else-if="isFieldType(selectValueType, model.values[0].values[0])"></b-form-select>
+      <b-form-select class="mr-1" v-model="localModel.values[0].values[0].op" :options="operators(localModel.values[0].values[0])"></b-form-select>
+      <b-form-input class="mr-1" type="text" v-model="localModel.values[0].values[0].value" v-if="isFieldType(substringValueType, localModel.values[0].values[0])"></b-form-input>
+      <pf-form-datetime class="mr-1" v-model="localModel.values[0].values[0].value" v-else-if="isFieldType(datetimeValueType, localModel.values[0].values[0])" :config="{useCurrent: true}"></pf-form-datetime>
+      <pf-form-prefix-multiplier class="mr-1" v-model="localModel.values[0].values[0].value" v-else-if="isFieldType(prefixmultipleValueType, localModel.values[0].values[0])"></pf-form-prefix-multiplier>
+      <b-form-select class="mr-1" v-localModel.lazy="localModel.values[0].values[0].value" :options="values(localModel.values[0].values[0])" v-else-if="isFieldType(selectValueType, localModel.values[0].values[0])"></b-form-select>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearchBoolean.vue]: `v-localModel`? Regexp glitch?

File: html/pfappserver/root/static.alt/src/components/pfSearchBoolean.vue [hunk 7]
```diff
@@ -36,16 +36,16 @@
               <b-form-input type="text" class="mr-1" v-model="rule.value" v-if="isFieldType(substringValueType, rule)"></b-form-input>
               <pf-form-datetime class="mr-1" v-model="rule.value" v-else-if="isFieldType(datetimeValueType, rule)" :config="{useCurrent: true}" :moments="['-1 hours', '-1 days', '-1 weeks', '-1 months', '-1 quarters', '-1 years']"></pf-form-datetime>
               <pf-form-prefix-multiplier class="mr-1" v-model="rule.value" v-else-if="isFieldType(prefixmultipleValueType, rule)"></pf-form-prefix-multiplier>
-              <b-form-select class="mr-1" v-model.lazy="rule.value" :options="values(rule)" v-else-if="isFieldType(selectValueType, rule)"></b-form-select>
-              <b-button class="ml-auto mr-1 nodrag" v-if="model.values.length > 1 || model.values[outerindex].values.length > 1 && drag === false" variant="link" v-b-tooltip.hover.left.d1000 :title="$t('Delete statement')" @click="removeStatement(outerindex, innerindex)"><icon name="trash-alt"></icon></b-button>
+              <b-form-select class="mr-1" v-localModel.lazy="rule.value" :options="values(rule)" v-else-if="isFieldType(selectValueType, rule)"></b-form-select>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearchBoolean.vue]: `v-localModel`? Regexp glitch?

File: html/pfappserver/root/static.alt/src/composables/useFormProvide.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      lastTickDebouncer = createDebouncer()
+    lastTickDebouncer({
+      handler: () => nextTick(() => {
+        lastTick.value = (new Date()).getTime()
+      }),
+      time: 100
+    })
+  }, { deep: true, immediate: true })
+  provide('lastTick', lastTick)
+  */
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/composables/useFormProvide.js]: Will this commented code be eventually uncommented?

**Review by cgx — CHANGES_REQUESTED** [review]

File: html/pfappserver/root/static.alt/package.json (hunk 3 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/package.json:63]: Existing dependency is not enough to use https://github.com/inverse-inc/packetfence/blob/2ed21aca981c35c0a36f78e43128be0dda289700/html/pfappserver/root/static.alt/vue.config.js#L40

Direct dep is needed.

**Pull request merged**